### PR TITLE
Use Path() where possible to cleanup the code a little

### PR DESCRIPTION
**Pull request by dirkmueller**

### CODE REVIEW

**Comment by dcermak** [comment]
Thanks, this LGTM! Please just run `poetry run ruff check --fix` to sort the imports.

File: src/bci_build/package/appcontainers.py [hunk 1]
```diff
@@ -218,10 +214,7 @@
     )
 
 
-with open(
-    os.path.join(os.path.dirname(__file__), "rmt", "entrypoint.sh")
-) as entrypoint:
-    _RMT_ENTRYPOINT = entrypoint.read(-1)
+_RMT_ENTRYPOINT = (Path(__file__).parent / "rmt" / "entrypoint.sh").read_bytes()
```
Comment (dcermak) [inline src/bci_build/package/appcontainers.py:217]:
```suggestion
_RMT_ENTRYPOINT = (Path(__file__).parent / "rmt" / "entrypoint.sh").read_text()
```

File: src/bci_build/package/appcontainers.py [hunk 2]
```diff
@@ -253,16 +246,10 @@
...
-
-with open(
-    os.path.join(os.path.dirname(__file__), "postgres", "LICENSE")
-) as license_file:
-    _POSTGRES_LICENSE = license_file.read(-1)
-
+_POSTGRES_ENTRYPOINT = (
+    Path(__file__).parent / "postgres" / "entrypoint.sh"
+).read_bytes()
+_POSTGRES_LICENSE = (Path(__file__).parent / "postgres" / "LICENSE").read_bytes()
```
Comment (dcermak) [inline src/bci_build/package/appcontainers.py:252]:
```suggestion
_POSTGRES_ENTRYPOINT = (
    Path(__file__).parent / "postgres" / "entrypoint.sh"
).read_text()
_POSTGRES_LICENSE = (Path(__file__).parent / "postgres" / "LICENSE").read_text()
```

File: src/bci_build/package/appcontainers.py [hunk 3]
```diff
@@ -395,9 +385,10 @@
 ]
 
 GRAFANA_FILES = {}
-for filename in {"run.sh", "LICENSE"}:
-    with open(os.path.join(os.path.dirname(__file__), "grafana", filename)) as cursor:
-        GRAFANA_FILES[filename] = cursor.read()
+for filename in ("run.sh", "LICENSE"):
+    GRAFANA_FILES[filename] = (
+        Path(__file__).parent / "grafana" / filename
+    ).read_bytes()
```
Comment (dcermak) [inline src/bci_build/package/appcontainers.py:391]:
```suggestion
    ).read_text()
```

File: src/bci_build/package/appcontainers.py [hunk 4]
```diff
@@ -445,8 +436,7 @@
     "30-tune-worker-processes.sh",
     "index.html",
 ):
-    with open(os.path.join(os.path.dirname(__file__), "nginx", filename)) as cursor:
-        _NGINX_FILES[filename] = cursor.read(-1)
+    _NGINX_FILES[filename] = (Path(__file__).parent / "nginx" / filename).read_bytes()
```
Comment (dcermak) [inline src/bci_build/package/appcontainers.py:439]:
```suggestion
    _NGINX_FILES[filename] = (Path(__file__).parent / "nginx" / filename).read_text()
```

File: src/bci_build/package/basecontainers.py [hunk 5]
```diff
@@ -250,11 +252,7 @@ def _get_minimal_kwargs(os_version: OsVersion):
     )
 
 
-with open(
-    os.path.join(os.path.dirname(__file__), "gitea-runner", "osc_checkout")
-) as osc_checkout_f:
-    OSC_CHECKOUT = osc_checkout_f.read(-1)
-
+OSC_CHECKOUT = (Path(__file__).parent / "gitea-runner" / "osc_checkout").read_bytes()
```
Comment (dcermak) [inline src/bci_build/package/basecontainers.py:255]:
```suggestion
OSC_CHECKOUT = (Path(__file__).parent / "gitea-runner" / "osc_checkout").read_text()
```

**Review by dcermak — CHANGES_REQUESTED** [review]
`read_bytes` is messing with the pcp readme, I think we should be using `read_text` instead

**Comment by dirkmueller** [comment]
well, that makes it a str rather than bytes, which is what the functions are supposed to be handling. even though I changed it everywhere, there's something special about the PCP README..

**Comment by dcermak** [comment]
I guess this is the culprit: https://github.com/SUSE/BCI-dockerfile-generator/blob/1e8a4900b58cb89410297dc7a9d15b6d78488519/src/bci_build/package/__init__.py#L1007

File: src/bci_build/package/__init__.py [hunk 6]
```diff
@@ -1004,6 +1005,8 @@ def readme_url(self) -> str:
     @property
     def readme(self) -> str:
         if "README.md" in self.extra_files:
+            if isinstance(self.extra_files["README.md"], bytes):
+                return self.extra_files["README.md"].decode('utf-8')
```
Comment (dcermak) [inline src/bci_build/package/__init__.py:1009]:
```suggestion
                return self.extra_files["README.md"].decode("utf-8")
```

**Comment by dcermak** [comment]
Thank you!

**Pull request merged**